crates/sui-core/src/consensus_handler.rs

```diff
@@ -16,37 +16,42 @@ use tracing::{debug, instrument, warn};
 pub struct ConsensusHandler {
     state: Arc<AuthorityState>,
     sender: mpsc::Sender<ConsensusListenerMessage>,
-    hash: Mutex<u64>,
+    last_seen: Mutex<ExecutionIndicesWithHash>,
 }
 
 impl ConsensusHandler {
     pub fn new(state: Arc<AuthorityState>, sender: mpsc::Sender<ConsensusListenerMessage>) -> Self {
-        let hash = Mutex::new(0);
+        let last_seen = Mutex::new(Default::default());
         Self {
             state,
             sender,
-            hash,
+            last_seen,
         }
     }
 
-    fn update_hash(&self, index: ExecutionIndices, v: &[u8]) -> ExecutionIndicesWithHash {
-        let mut hash_guard = self
-            .hash
+    fn update_hash(&self, index: ExecutionIndices, v: &[u8]) -> Option<ExecutionIndicesWithHash> {
```

Comment:
I believe this method now bears quite lots of critical logic. Could you please add a test for it?

Reply:
Added test

Reply:
Awesome thanks 👍 !